src/core/core.scale.js

```diff
@@ -350,19 +354,28 @@ var Scale = Element.extend({
 		me.beforeCalculateTickRotation();
 		me.calculateTickRotation();
 		me.afterCalculateTickRotation();
-		// Fit
+
 		me.beforeFit();
 		me.fit();
 		me.afterFit();
+
 		// Auto-skip
-		me._ticksToDraw = tickOpts.display && tickOpts.autoSkip ? me._autoSkip(me._ticks) : me._ticks;
+		me._ticksToDraw = tickOpts.display && tickOpts.autoSkip ? me._autoSkip(ticks) : ticks;
+
+		if (sampleSize) {
```

Comment:
Is there no way to do only call `me._convertTicksToLabels` once?

Reply:
If the concern is about performance, I think it isn't much of an issue because it's only called twice when the `sampleSize` option is provided and the overall performance savings is still really huge in that case (chart renders almost twice as fast in my test case as shown in PR description)

In terms of code cleanliness and maintainability, there's a lot of difficulty in changing much of the existing code to make this cleaner because so many of the methods and properties surrounding this are public that it's very hard to make too many changes while maintaining backwards compatibility. The other thing that makes it quite hard is the number of interdependencies between functions. I provided some explanation of why I did it this way in this comment: https://github.com/chartjs/Chart.js/pull/6508#issuecomment-530498200. I think in 3.0 we can clean this up some (e.g. remove `me.ticks`), but I haven't been able to find any better ways to do this than what's being done here while maintaining backward compatibility for the 2.x branch